addon/components/search-result/component.js

```diff
@@ -0,0 +1,174 @@
+import Ember from 'ember';
+import layout from './template';
+import Analytics from '../../mixins/analytics';
+import hostAppName from '../../mixins/host-app-name';
+
+/**
+ * Adapted from Ember-SHARE and Ember Preprints
+ * Used for search results on discover page.
+ *
+ * ```handlebars
+ * {{search-result
+ *      detailRoute=detailRoute
+ *      addFilter='addFilter'
+ *      result=result
+ *      queryParams=queryParams
+ *      filterReplace=filterReplace
+ *      updateFilters=(action 'updateFilters')
+ * }}
+ * ```
+ * @class search-result
+ */
+export default Ember.Component.extend(Analytics, hostAppName, {
+    layout,
+    maxTags: 10,
+    maxSubjects: 10,
+    maxCreators: 10,
+    maxDescription: 300,
+    showBody: false,
+    /**
+     * Array of query params being used in consuming app
+     * @property {Array} queryParams
+     */
+    queryParams: null,
+    /**
+     * Search result from SHARE
+     * @property {Object} result
+     */
+    result: null,
+    providerUrlRegex: {
+        //'bioRxiv': '', doesnt currently have urls
+        Cogprints: /cogprints/,
+        OSF: /https?:\/\/((?!api).)*osf.io/, // Doesn't match api.osf urls
+        PeerJ: /peerj/,
+        arXiv: /arxivj/,
+        'ClinicalTrials.gov': /http:\/\/clinicaltrials.gov/,
+    },
+    /**
+     * Name of detail route for consuming application, if you want search result to link to a route in the consuming spp
+     * @property {String} detailRoute
+     */
+    detailRoute: null,
+    footerIcon: Ember.computed('showBody', function() {
+        return this.get('showBody') ? 'caret-up' : 'caret-down';
+    }),
+
+    type: Ember.computed('result.type', function() {
+        return this.get('result.type').replace(/\w\S*/g, function(str) {return str.capitalize();});
+    }),
+    safeTitle: Ember.computed('result.title', function() {
+        return Ember.String.htmlSafe(this.get('result.title')).string;
+    }),
+    safeDescription: Ember.computed('result.description', function() {
+        return Ember.String.htmlSafe(this.get('result.description')).string;
```

Comment:
I very strongly recommend against using this method. Perhaps a better name should be `Ember.String.weReallyReallyTrustTheUserEnteredOkTextOnThisOneOtherwiseWhoopsie`.

The only time this should *ever* be used is for text from a trusted source that we completely control (like a string of advisory board members from the API). I suspect the description comes from some external input and therefore should not be marked as safe. This is especially true for SHARE, where the text we harvest comes from *many* external sources that all have different rules.

Can you clarify where this came in? I suspect the appropriate action will be a combination of the following:

1. Remove all uses of `htmlSafe` on user-entered text within this PR,
2. Send a note to whichever upstream repo this was borrowed from and make sure the fix gets propagated.
3. Identify the reason this was ever used, and develop alternatives to fit the intended use case. For example, most of our ember-data models clean up bad ampersands via a custom [field type](https://github.com/abought/ember-osf/blob/6627dbefd88fbefdc176a3ae300c45347b2cfd5c/addon/transforms/fixstring.js#L28-L28), also available as a [template helper](https://github.com/abought/ember-osf/blob/6627dbefd88fbefdc176a3ae300c45347b2cfd5c/addon/helpers/fix-special-char.js#L16-L16) for your convenience